README.md

```diff
@@ -1,95 +1,201 @@
+## Using Jigawatt
 
-### Properties
-Jigawatt Middleware must consist of at least one of three properties:
-- `awesomize`: `(Validator -> AwesomizeSpec) -> Request -> Object a`
-  - validates/sanitizes the request to initialize the `data` object
-  - _for more info on `awesomize`, see https://www.npmjs.com/package/awesomize_
+First, require the module.
 
-- `io`: `Request, Data -> Object a`
-  - returns an object that is merged into `req.data`
+`const JW = require('jigawatt')`
 
-- `transform`: `Request, Data -> Object a`
-  - returns an object to become the new `req.data` value
+### Basic Usage Example
+
+Consider the following endpoint:
 
-### Example Middleware
 ```
-const _         = require('ramda')
-const Bluebird  = require('bluebird')
+app.get('/order/:id', (req, res)=> {
+
+  const data = {
+    order    : db.getById(req.params.id)              // Assuming that
+  , product  : db.getProductByOrderId(req.params.id)  // this is data
+  , customer : db.getCustomerByOrderId(req.params.id) // from database
+  }
 
-const Order = require('../domain/order.js')
+  const getOrder = JW(formatOrder, data)
+      , result   = { json : (data) => data }
 
-const getById = {
+...
+```
+
+Jigawatt middleware functions, such as `formatOrder` in this case, are created to handle the incoming data:
+
+```
+const formatOrder = {
   awesomize: (v) => ({
-    order_id: {
-      read: _.path(['params', 'orderId'])
-    , validation: [ v.required ]
+    id : {
+      read : R.path([ 'order', 'id' ])
+    , validate : [ v.required, v.isInt ]
     }
+  , product  : { sanitize : [ R.trim ] }
+  , customer : { validate : [ v.required ] }
   })
 
-  io: (req, data) => {
-    return Bluebird.props({
-      order : Order.getById(data.order_id)
-    })
-  }
-};
-
-const uniteDetails = {
-  transform: (req, data) => {
-    return {
-      order     : data.order
-    , customer  : data.customer
-    , product   : data.product
-    , shipping  : data.shipping  
-    }
-  }
+, io: (req, data) => ({
+    orderId  : data.id
+  , product  : data.product
+  , customer : data.customer
+  })
 }
+```
+
+This will return a promise which, when resolved, will be a new beautified object.
 
-module.exports = {
-  getById
-, uniteDetails
+```
+...
+
+Promise
+  .resolve(
+    getOrder(data, result)
+  )
+  .then((orderDetails) => {
+    // do what you will with the data
+  })
+```
+**Example output:**
+```
+{ "orderId"  : "1234"
+, "product"  : "Mechanical pencil"
+, "customer" : "Jabroni Seagull"
 }
 ```
 
+This might be a bit overwhelming at first sight, so let's break it down in a bit more detail...
+
+### Middleware Structure
+
+Jigawatt Middleware expects at least one of the following three properties:
+
+#### awesomize
+**`awesomize :: (Validator -> AwesomizeSpec) -> Request -> Object a`**
+
+- Normalize/Sanitize/Validate an object
+- For more detailed documentation about awesomize, visit the [awesomize documentation](https://github.com/influentialpublishers/awesomize)
```

Comment:
You can just hot link to the awesomize documentation for each piece. If the awesomize documentation sucks then update that as well :D